crates/sui-indexer/src/models/objects.rs

```diff
@@ -252,12 +279,19 @@ impl TryFrom<StoredObject> for Object {
     type Error = IndexerError;
 
     fn try_from(o: StoredObject) -> Result<Self, Self::Error> {
-        bcs::from_bytes(&o.serialized_object).map_err(|e| {
-            IndexerError::SerdeError(format!(
-                "Failed to deserialize object: {:?}, error: {}",
-                o.object_id, e
-            ))
-        })
+        if let Some(serialized_object) = o.serialized_object {
+            let object = bcs::from_bytes(&serialized_object).map_err(|e| {
+                IndexerError::SerdeError(format!(
+                    "Failed to deserialize object: {:?}, error: {}",
+                    o.object_id, e
+                ))
+            })?;
+            return Ok(object);
+        }
+        Err(IndexerError::PersistentStorageDataCorruptionError(format!(
```

Comment:
Is it possible for this function to be called on a deleted/wrapped object? i.e. is this error an invariant violation that should never happen or an error that can happen and needs to be handled by the caller?